apis/kubedb/v1alpha1/percona_types.go

```diff
@@ -0,0 +1,97 @@
+package v1alpha1
+
+import (
+	"github.com/appscode/go/encoding/json/types"
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	mona "kmodules.xyz/monitoring-agent-api/api/v1"
+	ofst "kmodules.xyz/offshoot-api/api/v1"
+)
+
+const (
+	ResourceCodePercona     = "pc"
+	ResourceKindPercona     = "Percona"
+	ResourceSingularPercona = "percona"
+	ResourcePluralPercona   = "perconas"
+)
+
+// +genclient
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Percona defines a percona variation of Mysql database.
+type Percona struct {
+	metav1.TypeMeta   `json:",inline,omitempty"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              PerconaSpec   `json:"spec,omitempty"`
+	Status            PerconaStatus `json:"status,omitempty"`
+}
+
+type PerconaSpec struct {
+	// Version of Percona to be deployed.
+	Version types.StrYo `json:"version"`
+
+	// Number of instances to deploy for Percona
+	Replicas *int32 `json:"replicas,omitempty"`
+
+	// StorageType can be durable (default) or ephemeral
+	StorageType StorageType `json:"storageType,omitempty"`
+
+	// Storage spec to specify how storage shall be used.
+	Storage *core.PersistentVolumeClaimSpec `json:"storage,omitempty"`
+
+	// Database authentication secret
+	DatabaseSecret *core.SecretVolumeSource `json:"databaseSecret,omitempty"`
+
+	// Init is used to initialize database
+	// +optional
+	Init *InitSpec `json:"init,omitempty"`
+
+	// BackupSchedule spec to specify how database backup will be taken
+	// +optional
+	BackupSchedule *BackupScheduleSpec `json:"backupSchedule,omitempty"`
```

Comment:
We are going to deprecate `snapshot` from kubedb. So, we can skip `BackupSpec`. 

Other snapshot related fields (eg, `Init.snapshotSource`) will be removed later, so don't implement those(`snapshot` related) part in operator/controller.